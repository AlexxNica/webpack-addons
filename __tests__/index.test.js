'use strict';
const utils = require('../index');

describe('utils', () => {
	describe('arrowFunction', () => {
		it('should stringify an arrowFunction', () => {
			expect(utils.arrowFunction('app.js')).toMatchSnapshot();
		});
	});
	describe('regularFunction', () => {
		it('should stringify an arrowFunction', () => {
			expect(utils.regularFunction('app.js')).toMatchSnapshot();
		});
	});
	describe('dynamicPromise', () => {
		it('should stringify an single value', () => {
			expect(utils.dynamicPromise('app.js')).toMatchSnapshot();
		});
		it('should stringify an array', () => {
			expect(utils.dynamicPromise(['app.js', 'index.js'])).toMatchSnapshot();
		});
	});
	describe('assetFilterFunction', () => {
		it('should stringify an assetFilterFunction', () => {
			expect(utils.assetFilterFunction('js')).toMatchSnapshot();
		});
	});
	describe('externalRegExpFunction', () => {
		it('should stringify an externalRegExpFunction', () => {
			expect(utils.externalRegExpFunction('js')).toMatchSnapshot();
		});
	});
	describe('commonChunksPluginCreate', () => {
		it('should stringify an commonChunksPlugin', () => {
			expect(utils.commonChunksPluginCreate('vendor')).toMatchSnapshot();
		});
	});
	describe('createRequire', () => {
		it('should stringify an require statement', () => {
			expect(utils.createRequire('webpack')).toMatchSnapshot();
		});
	});
	describe('Inquirer', () => {
		it('should make an List object', () => {
			expect(utils.List('entry', 'does it work?', ['Yes', 'Maybe'])).toEqual({
				type: 'list',
				name: 'entry',
				message: 'does it work?',
				choices: ['Yes', 'Maybe']
			});
		});
		it('should make an RawList object', () => {
			expect(utils.RawList('output', 'does it work?', ['Yes', 'Maybe'])).toEqual({
				type: 'rawlist',
				name: 'output',
				message: 'does it work?',
				choices: ['Yes', 'Maybe']
			});
		});
		it('should make an CheckList object', () => {
			expect(utils.CheckList('context', 'does it work?', ['Yes', 'Maybe'])).toEqual({
				type: 'checkbox',
				name: 'context',
				message: 'does it work?',
				choices: ['Yes', 'Maybe']
			});
		});
		it('should make an Input object', () => {
			expect(utils.Input('plugins', 'what is your plugin?')).toEqual({
				type: 'input',
				name: 'plugins',
				message: 'what is your plugin?'
			});
		});
	});
});
