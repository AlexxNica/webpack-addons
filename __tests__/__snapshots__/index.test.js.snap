// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils arrowFunction should stringify an arrowFunction 1`] = `"() => 'app.js'"`;

exports[`utils assetFilterFunction should stringify an assetFilterFunction 1`] = `
"function (assetFilename) {
 return assetFilename.endsWith('.js');
}"
`;

exports[`utils commonChunksPluginCreate should stringify an commonChunksPlugin 1`] = `"new webpack.optimize.CommonsChunkPlugin({name:'vendor',filename:'vendor-[hash].min.js'})"`;

exports[`utils createRequire should stringify an require statement 1`] = `"const webpack = require('webpack');"`;

exports[`utils dynamicPromise should stringify an array 1`] = `"() => new Promise((resolve) => resolve(['app.js','index.js']))"`;

exports[`utils dynamicPromise should stringify an single value 1`] = `"() => new Promise((resolve) => resolve('app.js'))"`;

exports[`utils externalRegExpFunction should stringify an externalRegExpFunction 1`] = `
"
 function (context, request, callback) {
 if (/js/.test(request)){
   return callback(null, 'commonjs' + request);
}
callback();
}"
`;

exports[`utils regularFunction should stringify an arrowFunction 1`] = `
"function () {
 return 'app.js'
}"
`;
